include/ginkgo/core/base/array.hpp

```diff
@@ -550,6 +550,60 @@ class array {
         return *this;
     }
 
+    /**
+     * Copies or converts data from a const_array_view.
+     *
+     * In the case of an array target, the array is resized to match the
+     * source's size. In the case of a view target, if the dimensions are not
+     * compatible a gko::OutOfBoundsError is thrown.
+     *
+     * This does not invoke the constructors of the elements, instead they are
+     * copied as POD types.
+     *
+     * The executor of this is preserved. In case this does not have an assigned
+     * executor, it will inherit the executor of other.
+     *
+     * @param other  the const_array_view to copy from
+     * @tparam OtherValueType  the value type of `other`
+     *
+     * @return this
+     */
+    template <typename OtherValueType>
+    std::enable_if_t<std::is_convertible<OtherValueType, ValueType>::value,
+                     array>&
+    operator=(const detail::const_array_view<OtherValueType>& other)
```

Comment:
Do we need the convertible aspect? That's a part of `array` that I regret adding, because it mixes a very basic data structure with kernel calls, so it makes untangling the library harder.

Reply:
I'm fine with disabling it. I don't need it here.